pkg/doc/verifiable/credential_jwt.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/square/go-jose/v3"
+	"github.com/square/go-jose/v3/jwt"
+)
+
+// JWTCredClaims is JWT Claims extension by Verifiable Credential.
+type JWTCredClaims struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// MarshalJWS serializes JWT into signed form (JWS)
+func (vcc *JWTCredClaims) MarshalJWS(signatureAlg JWTAlgorithm, privateKey interface{}, keyID string) (string, error) { //nolint:lll
+	key := jose.SigningKey{Algorithm: signatureAlg.Jose(), Key: privateKey}
+
+	var signerOpts = &jose.SignerOptions{}
+	signerOpts.WithType("JWT")
+	signerOpts.WithHeader("kid", keyID)
+
+	signer, err := jose.NewSigner(key, signerOpts)
+	if err != nil {
+		return "", fmt.Errorf("failed to create signer: %w", err)
+	}
+
+	// create an instance of Builder that uses the signer
+	builder := jwt.Signed(signer).Claims(vcc)
+
+	// validate all ok, sign with the key, and return a compact JWT
+	jws, err := builder.CompactSerialize()
+	if err != nil {
+		return "", fmt.Errorf("failed to sign JWT: %w", err)
+	}
+
+	return jws, nil
+}
+
+// MarshalUnsecuredJWT serialized JWT into unsecured JWT.
+func (vcc *JWTCredClaims) MarshalUnsecuredJWT() (string, error) {
+	headers := map[string]string{
+		"alg": "none",
+	}
+	return marshalUnsecuredJWT(headers, vcc)
+}
+
+// jwtCredObjClaims is used to get generic JSON object content of "vc" claim of JWT.
+type jwtCredObjClaims struct {
+	CredObj map[string]interface{} `json:"vc,omitempty"`
+}
+
+func mergeRefinedVC(jsonCred map[string]interface{}, rawCred *rawCredential) error {
+	rawData, err := json.Marshal(rawCred)
+	if err != nil {
+		return err
+	}
+
+	var rawMap map[string]interface{}
+
+	err = json.Unmarshal(rawData, &rawMap)
+	if err != nil {
+		return err
+	}
+
+	// make the merge
+	for k, v := range rawMap {
+		jsonCred[k] = v
+	}
+
+	return nil
+}
+
+// credJWTDecoder parses JWT claims from serialized token into JWTCredClaims struct and JSON object.
+// The implementation depends on the type of serialization, e.g. signed (JWT), unsecured (plain JWT).
+type credJWTDecoder interface {
+	CredClaims(rawJwt []byte) (*JWTCredClaims, error)
+	JSONObj(rawJwt []byte) (map[string]interface{}, error)
```

Comment:
I would rename JSONObj.

Reply:
Cannot find a better naming... Could you advise?

Reply:
I would convey the idea that you are unmarshaling (parsing) the VC field vs the JWT token that has a vc field.

Something like UnmarshalVC and the other is UnmarshalVCToken? Perhaps you have something better :).

Reply:
I have `UnmarshalClaims()` for the first and `UnmarshalVCClaim()` for the second.